clang/test/CXX/drs/cwg27xx.cpp

```diff
@@ -1,11 +1,23 @@
-// RUN: %clang_cc1 -triple x86_64-linux-gnu -std=c++98 -verify=expected %s
+// RUN: %clang_cc1 -triple x86_64-linux-gnu -std=c++98 -Wgnu-folding-constant -verify=expected %s
```

Comment:
I hope `-Wgnu-folding-constant` is not going to be needed after you apply other suggestions. Even if it's needed, this has to be handled some other way, not with an additional flag.